overlord/servicestate/servicestate_test.go

```diff
@@ -220,7 +221,7 @@ NeedDaemonReload=no
 		err = sd.DecorateWithStatus(app, snapApp)
 		c.Assert(err, IsNil)
 		c.Check(app.Active, Equals, enabled)
-		c.Check(app.Enabled, Equals, enabled)
+		c.Check(app.Enabled, Equals, true)
```

Comment:
It's possible this might be a separate issue in itself. I have made a note of this to investigate this specific scenario. Currently the whole handling around activated services is a bit hit'n'miss.